README.md

```diff
@@ -32,22 +32,29 @@ Here's a simple example of creating a provider:
 
 Probes may be fired via the provider object:
 
-    dtp.fire("probe1", function(p) {
+    dtp.fire("probe1", function() {
         return [1, 2];
     });
-    dtp.fire("probe2", function(p) { 
+    dtp.fire("probe2", function() { 
         return ["hello, dtrace via provider", "foo"];
     });
 
 or via the probe objects themselves:
 
-    p1.fire(function(p) {
+    p1.fire(function() {
       return [1, 2, 3, 4, 5, 6];
     });
-    p2.fire(function(p) {
+    p2.fire(function() {
       return ["hello, dtrace via probe", "foo"];
     });
 
+You can pass arguments to the callback after it, similar to `setImmediate`:
```

Comment:
I think it might be worth explaining why you would (or wouldn't) do this.  This is sort of mentioned at line 72 below (in the new version), but maybe something like:

> fire() uses a callback that returns the arguments to be provided when the DTrace probe actually fires. This indirection allows you to call fire() unconditionally when you want to fire the probe, but the callback will be invoked to construct the arguments only if the DTrace probe is actually enabled.  (The probe can be enabled and disabled at any time in response to users using the dtrace(1M) command.)  This in turn allows probes to take arguments that would be expensive to construct -- without having to construct those arguments unless the probe is actually enabled.
>
> In some cases, this approach encourages creating a new closure at each probe site (in order to pass it to fire()).  For extremely CPU-intensive or memory-conscious workloads, this overhead can be avoided by using a statically-defined function (i.e., one not created with each call to fire()) and supplying arguments to that function as additional arguments to fire().  Be careful to avoid using this pattern with arguments that are themselves expensive to construct, as that undermines the design goal of minimizing the effect of disabled probes.

Feel free to take that or rewrite it, but that's the kind of information I think might be useful to include.